### PR TITLE
ON-6365 # Added jwtverify function

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: lib/tenant.js [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-  ONEBLINK: 'https://auth-api.blinkm.io',
-  CIVICPLUS: 'https://auth-api.transform.civicplus.com',
+const tenants = {
+  ONEBLINK: {
+    url: 'https://auth-api.blinkm.io',
+    iss:
+      'https://cognito-idp.ap-southeast-2.amazonaws.com/ap-southeast-2_o1t3ntGWx',
+  },
+  CIVICPLUS: {
+    url: 'https://auth-api.transform.civicplus.com',
```
Comment (mymattcarroll) [inline lib/tenant.js:11]: can we change this property to `apiOrigin` please.

File: lib/verify-jwt.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+const getPublicKey = (
+  iss /* : string */,
+  kid /* : string */,
+) /* : Promise<string> */ => {
+  const jwksClient = new JwksClient({
+    cache: true,
+    jwksUri: `${iss}/.well-known/jwks.json`,
+  })
```
Comment (mymattcarroll) [inline lib/verify-jwt.js:25]: This needs to be created outside this function to allow for caching.

File: lib/verify-jwt.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  return new Promise((resolve, reject) => {
+    jwksClient.getSigningKey(kid, (err, key) => {
+      if (err) return reject('Error retreiving public key.')
```
Comment (mymattcarroll) [inline lib/verify-jwt.js:28]:
```suggestion
      if (err) return reject(err)
```

File: classes/Forms.js [hunk 4]
```diff
@@ -8,14 +8,14 @@ const OneBlinkAPI = require('../lib/one-blink-api.js')
 const setPreFillData = require('../lib/pre-fill-data')
 const { validateWithFormSchema } = require('../lib/forms-validation.js')
 const generateFormElement = require('../lib/generate-form-element.js')
-const getTenantUrl = require('../lib/tenant')
 module.exports = class Forms extends OneBlinkAPI {
   constructor(options /* : ConstructorOptions */) {
     options = options || {}
     super(
-      getTenantUrl(options.tenant, options.oneBlinkAPIOrigin),
+      options.oneBlinkAPIOrigin,
```
Comment (mymattcarroll) [inline classes/Forms.js:15]: Can't remove this now? And from all the other class constructors?

File: lib/tenant.js [hunk 5]
```diff
@@ -1,23 +1,25 @@
...
+    iss: 'https://cognito-idp.us-east-2.amazonaws.com/us-east-2_A92OPccYd',
+  },
 }
```
Comment (mymattcarroll) [inline lib/tenant.js:14]: I thought we were updating this to allow for a `ONEBLINK_SDK_ENVIRONMENT` environment variable that we could set to `test` internally when we want to test the SDK?

File: lib/tenant.js (hunk 1 above)
Comment (Zaxist) [reply lib/tenant.js:11]: yep

File: lib/verify-jwt.js (hunk 2 above)
Comment (Zaxist) [reply lib/verify-jwt.js:25]: Do you mean in the function calling it?

Comment (mymattcarroll) [reply lib/verify-jwt.js:25]: I mean it needs to move to line 17. Can can discuss in person if you'd like.

File: classes/Forms.js (hunk 4 above)
Comment (Zaxist) [reply classes/Forms.js:15]: Didn't put the env var feature in yet

File: lib/tenant.js (hunk 5 above)
Comment (Zaxist) [reply lib/tenant.js:14]: We discussed it

File: lib/verify-jwt.js (hunk 2 above)
Comment (Zaxist) [reply lib/verify-jwt.js:25]: It needs the iss property though?

File: docs/forms-apps.md [hunk 6]
```diff
@@ -27,10 +28,54 @@ const OneBlink = require('@oneblink/sdk')
...
 ```
 
+## `verifyJWT()`
+
+Verify a JWT and return its result
+
+### Example
+
+```javascript
+formsAppsSDK
```
Comment (mymattcarroll) [inline docs/forms-apps.md:43]: Can we also support passing in the `Authorization` header value which will be `Authorization: 'Bearer {token}'`

Would be good to have that or here in the example aswell.

```js
const token = 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c'
// or
const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c'
formsAppsSDK
  .verifyJWT(token)
```

**Pull request merged**